gridstatus/spp.py

```diff
@@ -213,17 +217,59 @@ def get_solar_and_wind_forecast_short_term(self, date, verbose=False):
         Returns:
             pd.DataFrame: forecast as dataframe.
         """
-        # The SHORT_TERM forecast is delayed up to 2 minutes.
+        # The short_term forecast is delayed up to 2 minutes.
         buffer_minutes = 2
 
+        now = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
         if date in ["latest", "today"]:
-            date = pd.Timestamp.now(tz=self.default_timezone) - pd.Timedelta(
-                minutes=buffer_minutes,
-            )
+            date = now
 
-        return self._get_solar_and_wind_forecast_short_term(date, verbose=verbose)
+        date = utils._handle_date(date, self.default_timezone)
+        date = min(now, date)
+
+        if end:
+            end = utils._handle_date(end, self.default_timezone)
+            # Make sure we don't request files that don't exist yet
+            end = min(now, end)
+
+        df = self._retrieve_solar_and_wind_forecast_short_term(
+            date,
+            end,
+            verbose,
+        )
 
-    def get_solar_and_wind_forecast_mid_term(self, date, verbose=False):
+        df["Forecast Type"] = "SHORT_TERM"
+
+        if end:
```

Comment:
is it possible for this condition to actually be met since you are not requesting files after `end`?